go/cmd/dolt/commands/schcmds/import.go

```
@@ -16,26 +16,313 @@ package schcmds
+			ColMapper:      colMapper,
+			FloatThreshold: floatThreshold,
+			KeepTypes:      apr.Contains(keepTypesParam),
+		},
+	}
+
+	sch, verr := inferSchemaFromFile(ctx, dEnv.DoltDB.ValueReadWriter().Format(), pks, &impArgs)
+
+	if verr != nil {
+		return verr
+	}
+
+	cli.Println(sql.SchemaAsCreateStmt(tblName, sch))
+
+	if !apr.Contains(dryRunFlag) {
+		schVal, err := encoding.MarshalAsNomsValue(context.Background(), root.VRW(), sch)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to encode schema.").AddCause(err).Build()
+		}
+
+		m, err := types.NewMap(ctx, root.VRW())
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to create table.").AddCause(err).Build()
+		}
+
+		tbl, err = doltdb.NewTable(ctx, root.VRW(), schVal, m)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to create table.").AddCause(err).Build()
+		}
+
+		root, err = root.PutTable(ctx, tblName, tbl)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to add table.").AddCause(err).Build()
+		}
+
+		err = dEnv.UpdateWorkingRoot(ctx, root)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to update the working set.").AddCause(err).Build()
+		}
+
+		cli.PrintErrln(color.CyanString("Created table successfully."))
+	}
+
+	return nil
+}
+
+func inferSchemaFromFile(ctx context.Context, nbf *types.NomsBinFormat, pkCols []string, args *importArgs) (schema.Schema, errhand.VerboseError) {
+	if args.fileType[0] == '.' {
+		args.fileType = args.fileType[1:]
+	}
+
+	var rd table.TableReadCloser
+	switch args.fileType {
+	case "csv":
+		f, err := os.Open(args.fileName)
+
+		if err != nil {
+			return nil, errhand.BuildDError("error: failed to open '%s'", args.fileName).Build()
+		}
+
+		defer f.Close()
+
+		rd, err = csv.NewCSVReader(nbf, f, csv.NewCSVInfo().SetDelim(args.delim))
+
+		if err != nil {
+			return nil, errhand.BuildDError("error: failed to create a CSVReader.").AddCause(err).Build()
+		}
+
+		defer rd.Close(ctx)
+	}
```

Probably should throw an error here on unsupported file types.

---

fixed